tensorboard/components/tf_dashboard_common/data-loader-behavior.ts

```diff
@@ -55,6 +55,25 @@ export const DataLoaderBehavior = {
      */
     loadDataCallback: Function,
 
+    // A function that takes a datum as argument and makes the HTTP
+    // request to fetch the data associated with the datum. It should return
+    // a promise that either fullfills with the data or rejects with an error.
+    // If the function doesn't bind 'this', then it will reference the element
+    // that includes this behavior.
+    // The default implementation calls this.requestManager.request with
+    // the value returned by this.getDataLoadUrl(datum) (see below).
+    // The only place getDataLoadUrl() is called is in the default
+    // implementation of this method. So if you override this method with
+    // an implementation that doesn't call getDataLoadUrl, it need not be
+    // provided.
+    requestData: {
```

Comment:
This method sound fair but it feels redundant to requestManager (which I forgot to specify as one of the prop here). Is your requirement so that you have to use different HTTP method depending on a datum?

It's cool If above question is true, but now, it feels odd to pass `getDataLoadUrl`. It feels like `requestData` should know how to form necessary url/payload/whatever with the datum we pass. What do you think?

Reply:
Thanks for the review, Stephan.

The requirement is that I need to do an HTTP POST (with the request data in the body) instead of GET (with the request data in the URL) to fetch the scalar values in tf-scalar-card (which uses tf-line-chart-data-loader which inherits the behavior in question). This happens independently of the datum. 

I agree with you that having both getDataLoadUrl and requestData is redundant. In fact, initially, I wanted to just have requestData and remove getDataLoadUrl, but that resulted in a change in interface that required changing quite a few other depending files; so I was reluctant.

However, if you agree with the new interface, I can amend this PR to do it. Basically, we'll have requestData as the function in data-loader-behavior, and remove getDataLoadUrl. I can then add an external function (in the tf-dashboard-common namespace) like:

function HTTPGetRequestDataFn(datumToUrlFn, requestManager)  {
  return (datum)=>requestManager.request(datumToUrlFn(datum));
}

so that users of the behavior can use it to specify the value of 'requestData' to get the old interface. This change would also decouple the data-loader-behavior from requestManager.

Let me know if you agree.

Reply:
@erzel I do agree to the new interface for its more genericness and I do think it is the right feature. 

Question though: if HTTP method does not change and just has to be POST as opposed to GET, you can just supply a new RequestManager that uses HTTP POST in the `request` method, right? That might unblock you immediately while we are migrating to adopt the new interface.

Reply:
Thanks for the suggestion. I'm tempted to do it. The only problem I see is that I'll need to encode the request to the POST method in the url and then decode it in my new request manager's 'request' method, which is a little hacky.

Reply:
@erzel ah right. I presume it uses body of the POST request as opposed to some query parameter? In that case, I guess we should move forward with the new interface and migrate other components over.

Reply:
Thanks! I was going to offer to make the changes to the other components myself, but I'm happy to not have to do this. Still if you like me to do it in this PR, I can. What do you think?

Reply:
I will do the refactor. Thanks! 😄